app/src/main/java/io/gnosis/safe/ui/base/Adapter.kt

```diff
@@ -44,11 +44,11 @@ abstract class Adapter<T, VH : Adapter.ViewHolder<T>> : RecyclerView.Adapter<VH>
     }
 
     data class Data<out T>(
```

Comment:
what is the intention of the Data class?

Reply:
I think it's a form of change dispatcher. I haven't looked at it closely, it's from the old app.

Reply:
I think we do not need it. Also, we would we want to expose diff result. ParentId is also something very special and should not be generalized.

Reply:
So the scope of this PR was to define the adapter and viewHolders for TX list. I don't want to start refactoring code that is from the legacy app in here. This Adapter is also extended by the `LifecycleAdapter` so I am sorry but I don't want this to blow up like other tasks have. If we want to get rid of this, then I say we do so in a different issue.

Reply:
It's not about blowing things up, but about not having code that is not needed. This just leads to technical debt and over-engineering. You can create an issue for this (Base Adapter, DiffUtils), link it in this pr, and merge. Everything else looks good.